using-mirrord/db-branching.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "DB Branching"
+description: "How to use mirrord to spin up isolated DB branch for safe development and testing DB migrations"
+date: 2025-08-31T00:00:00+03:00
+lastmod: 2025-08-31T00:00:00+03:00
+draft: false
+menu:
+  docs:
+    parent: "using-mirrord"
+toc: true
+tags: ["team", "enterprise"]
+---
+
+
+The `db_branches` feature in mirrord lets developers spin up isolated DB branch that mirror the remote DB, while running safely in isolation. This allows schema changes, migrations, and experiments without impacting teammates or shared environments.
+Currently, the feature is limited to MySQL databases and does not support schema or data replication.
+
+
+**When is this useful?**
+
+1. **Running schema migrations safely**  
+   Developers can test migrations and schema changes without risking corruption of the remote databases.
+
+2. **Experimenting locally**  
+   Developers get a branch of the remote DB connected to their local app automatically - no admin setup required.
+
+3. **Collaborating with teammates**  
+   If a branch already exists (with the same ID), mirrord reuses it. Developers can share branches as needed.
+
+   --- 
+
+## Prerequisites
+
+Before you start, make sure you have:  
+1. A MySQL database configured in your remote environment.  
+2. Your local application is using environment variables to store DB connection strings.  
+3. mirrord installed and working.  
+
+
+## Configuring `mirrord db branches`
+Developers define branches in their `mirrord.json`:
+```json
+{
+  "db_branches": [
+    {
+      "id": "users-mysql-db", // Optional
+      "type": "mysql",
+      "version": "8.0",
+      "name": "users-database-name",   // Optional
+      "ttl_secs": 60,                 // Optional
+      "connection": {
+        "url": { 
+            "type": "env", 
+            "variable": "DB_CONNECTION_URL"  // Required
+        }
+      }
+    }
+  ]
+}
+```
+Key Fields
+1. id: If reused, mirrord reattaches to the same branch. Can be used sharing database branch.
+2. type: Currently only "mysql" is supported.
+3. version: Database engine version.
+4. name: Remote database name to clone, the override URL uses 'name' so the connection URL looks like .../dbname.
+If name is dropped, override URL points just to the MySQL server; app must select DB manually.
+5. ttl_secs: Override for branch TTL (default is 30 minutes).
+6. connection.url: Required. The environment variable that contains your DB connection string.
+
+## Running With DB Branches
+
+1. Run your app 'mirrord exec' with mirrord.json configured as defined above:
```

Comment:
```suggestion
1. Run your app with mirrord and set the `db_branches` field in [the mirrord configuration file](https://metalbear.com/mirrord/docs/config).
```

That sentence is doesn't really work grammatically, and also users can run with the IDE, not only the CLI, `mirrord exec` is too specific. The colon at the end doesn't fit there, unless an introduction of a something comes before of it, and the other bullets are not in the same hierarchy level as this sentence.